Services/Administration/OJS.Services.Administration.Business/Participants/Validators/ChangeParticipationTimeValidator.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace OJS.Services.Administration.Business.Participants.Validators;
+
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using OJS.Common.Enumerations;
+using OJS.Services.Administration.Data;
+using OJS.Services.Administration.Models.Participants;
+using OJS.Services.Common.Validation;
+using System.Threading.Tasks;
+
+public class ChangeParticipationTimeValidator : BaseValidator<ChangeParticipationTimeModel>
+{
+    private readonly IContestsDataService contestsDataService;
+
+    public ChangeParticipationTimeValidator(
+        IContestsDataService contestsDataService)
+    {
+        this.contestsDataService = contestsDataService;
+
+        this.RuleFor(model => model.ContestId)
+            .Must((model, _) => ContestIdMustBeValid(model.ContestId))
+            .WithMessage("Invalid contest id.");
+
+        this.RuleFor(model => model.ContestId)
+            .MustAsync(async (contestId, _) => await this.ContestMustExist(contestId))
+            .WithMessage("The contest was not found.");
+
+        this.RuleFor(model => model.ContestId)
+            .MustAsync(async (contestId, _) => await this.ContestMustBeActive(contestId))
+            .WithMessage("The contest must be active.");
+
+        this.RuleFor(model => model.ContestId)
+            .MustAsync(async (contestId, _) => await this.ContestMustBeAnOnlineExam(contestId))
+            .WithMessage("The contest must be an online exam.");
+
+        this.RuleFor(model => model.ContestId)
+            .MustAsync(async (contestId, _) => await this.ContestDurationMustBeSet(contestId))
+            .WithMessage("The contest's duration has not been set.");
```

Comment:
Can you combine all those into a `CustomAsync` check, so that you can reduce the db queries.

Reply:
Done.